compiler/rustc_resolve/src/def_collector.rs

```diff
@@ -352,4 +369,29 @@ impl<'a, 'b> visit::Visitor<'a> for DefCollector<'a, 'b> {
             visit::walk_crate(self, krate)
         }
     }
+
+    fn visit_assoc_constraint(&mut self, constraint: &'a AssocConstraint) {
+        /*if let ImplTraitContext::UniversalInDyn(item_def) = self.impl_trait_context {
```

Comment:
This version is correct.

Reply:
We should only create the definition is `constraint.kind` is `AssocConstraintKind::Bound`.
`dyn Foo<T: Trait>` becomes `dyn Foo<T = impl Trait>`.